rpc/core/blocks.go

```diff
@@ -237,7 +237,29 @@ func padBytes(byt []byte, length int) ([]byte, error) {
 	return tmp, nil
 }
 
-// EncodeDataRootTuple takes a height and a data root and returns the equivalent of
+// To32PaddedHexBytes takes a number and returns its hex representation padded to 32 bytes.
+// Used to mimic the result of `abi.encode(number)` in Ethereum.
+func To32PaddedHexBytes(number uint64) ([]byte, error) {
```

Comment:
I think `1` is preferable because callers must explicitly check if the argument they use to call this function overflows a uint32. I think `2` is less preferable because it may result in a runtime error.

Reply:
ca0aa9524983f3f3a5a1776c6938be5352905272

Reply:
@rootulp Just thought about it again, we can actually pad a `uint64` to 32 bytes. The maximum number for `uint64` is `ffffffffffffffff`, am I missing something? 
I think we're mixing `uint64`  and `uint256` (also 256 can be represented in 32 bytes, but it's the maximum)

Reply:
🤦‍♂️ you're right, I totally messed that up. A uint64 is 64 bits which is 8 bytes which can definitely fit in 32 bytes I think the change can be reverted. Sorry for my mix up.

Reply:
cool, thanks